machines/bleShare/scan/scanMachine.ts

```diff
@@ -679,8 +680,21 @@ export const scanMachine =
           },
         },
         checkingLocationService: {
-          initial: 'checkingPermissionStatus',
+          initial: 'checkLocationService',
```

Comment:
Can we rename the parent state name / child state name accordingly as we have similar name for both states

Reply:
parent state is named as check**ing**LocationService as it is continuing operation, we named the child as checkLocationService, as it is a operation that is performed currently.

which gives more clarity, pls let me know if you have any other concerns.

Reply:
The state checkLocationService checks if location is turned on or not, where as checkingLocationService is combination of states for check if location turned on -> if yes check if app has permission to access location -> if yes go ahead or request.
Considering this , the parent is performing a more functionality than just checking for location service. Should be bring in more generic state name for the parent considering its functionalities? (like checkLocationStatus)

Reply:
updated it @KiruthikaJeyashankar